app/main.py

```diff
@@ -1,2 +1,67 @@
+from typing import Any, Optional, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length: int = 0
+        self.capacity: int = 8
+        self.hash_table: list[Optional[tuple]] = [None] * self.capacity
+
+    def _get_key_value(self, key: Hashable) -> Any:
```

Comment:
You're recalculating the hash value for the key multiple times, both in _get_key_value and _set_key_value as well as inside the tuple (key, hash(key), value). This can be optimized by calculating the hash once and passing it around where necessary.